homeassistant/components/sensor/envirophat.py

```diff
@@ -0,0 +1,195 @@
+"""
+Support for Enviro pHAT sensors.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.envirophat
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (TEMP_CELSIUS, CONF_DISPLAY_OPTIONS, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['envirophat==0.0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'envirophat'
+CONF_USE_LEDS = 'use_leds'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+SENSOR_TYPES = {
+    'light': ['light', ' ', 'mdi:weather-sunny'],
+    'light_red': ['light_red', ' ', 'mdi:invert-colors'],
+    'light_green': ['light_green', ' ', 'mdi:invert-colors'],
+    'light_blue': ['light_blue', ' ', 'mdi:invert-colors'],
+    'accelerometer_x': ['accelerometer_x', 'G', 'mdi:earth'],
+    'accelerometer_y': ['accelerometer_y', 'G', 'mdi:earth'],
+    'accelerometer_z': ['accelerometer_z', 'G', 'mdi:earth'],
+    'magnetometer_x': ['magnetometer_x', ' ', 'mdi:magnet'],
+    'magnetometer_y': ['magnetometer_y', ' ', 'mdi:magnet'],
+    'magnetometer_z': ['magnetometer_z', ' ', 'mdi:magnet'],
+    'temperature': ['temperature', TEMP_CELSIUS, 'mdi:thermometer'],
+    'pressure': ['pressure', 'hPa', 'mdi:gauge'],
+    'voltage_0': ['voltage_0', 'V', 'mdi:flash'],
+    'voltage_1': ['voltage_1', 'V', 'mdi:flash'],
+    'voltage_2': ['voltage_2', 'V', 'mdi:flash'],
+    'voltage_3': ['voltage_3', 'V', 'mdi:flash'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DISPLAY_OPTIONS, default=SENSOR_TYPES):
+        [vol.In(SENSOR_TYPES)],
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_USE_LEDS, default=False): cv.boolean
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
```

Comment:
We shouldn't setup the platform if the physical hardware isn't attached. I doubt that it's possible to attach the Hat when the Raspberry Pi is already running. Correct me if I wrong.

The users will end-up with a non-working integration in their setup. Log an error and make the setup fail.